dispatch.go

```diff
@@ -89,22 +90,35 @@ func newDispatch(enc encoder, dec byteReadingDecoder, l LogInterface, wef WrapEr
 
 type call struct {
 	ch             chan error
+	doneCh         chan struct{}
 	method         string
 	seqid          int
 	arg            interface{}
 	res            interface{}
 	errorUnwrapper ErrorUnwrapper
 	profiler       Profiler
+	ctx            context.Context
 }
 
-func newCall(m string, arg interface{}, res interface{}, u ErrorUnwrapper, p Profiler) *call {
+func newCall(ctx context.Context, m string, arg interface{}, res interface{}, u ErrorUnwrapper, p Profiler) *call {
 	return &call{
 		ch:             make(chan error),
+		doneCh:         make(chan struct{}),
 		method:         m,
 		arg:            arg,
 		res:            res,
 		errorUnwrapper: u,
 		profiler:       p,
+		ctx:            ctx,
+	}
+}
+
+func (c *call) Done(err error) {
+	// Ensure we only send a response and close doneCh once
+	select {
+	case c.ch <- err:
```

Comment:
As long as there's someone waiting on the other end, we won't drop the error. That said, you're correct, we should have a buffer of size 1 here because that avoids the race if the call finishes before the consumer gets blocked on the channel (extremely unlikely but possible)